### PR TITLE
fix(api): Use snuba connection pool when creating/deleting subscriptions.

### PR DESCRIPTION
**Pull request by wedamija**
We can't use `safe_urlopen` since it blacklists local services.

### CODE REVIEW

File: src/sentry/incidents/logic.py [hunk 1]
```diff
@@ -39,6 +38,7 @@
 from sentry.incidents import tasks
 from sentry.utils.committers import get_event_file_committers
 from sentry.utils.snuba import (
+    _snuba_pool,
```
Comment (markstory) [inline src/sentry/incidents/logic.py]: Importing things with `_` makes me 🤔 Should we make this a non-private symbol or add a better API?

Comment (BYK) [reply src/sentry/incidents/logic.py]: I'd love to get this merged first to unbreak the tests and prod and then follow up with improvements if that fits your bill.

Just pushed a merge from master, fixing the `black` conflicts.

Comment (wedamija) [reply src/sentry/incidents/logic.py]: @markstory I don't love it either, but I think we need to move the snuba stuff around a bit in general. The new `snuba` app I added in https://github.com/getsentry/sentry/pull/14383 could be a decent place.

**Review by markstory — APPROVED** [review]

File: src/sentry/incidents/logic.py [hunk 2]
```diff
@@ -755,24 +761,26 @@ def create_snuba_subscription(project, dataset, query, aggregations, time_window
     :return: A uuid representing the subscription id.
     """
     # TODO: Might make sense to move this into snuba if we have wider use for
-    # it.
-    resp = safe_urlopen(
-        settings.SENTRY_SNUBA + "/subscriptions",
+    # it.'
```
Comment (BYK) [inline src/sentry/incidents/logic.py]: Nit: dangling `'`

File: src/sentry/incidents/logic.py [hunk 3]
```diff
@@ -755,24 +761,26 @@ def create_snuba_subscription(project, dataset, query, aggregations, time_window
...
+                "aggregates": [alert_aggregation_to_snuba[agg] for agg in aggregations],
+                "time_window": time_window,
+                "resolution": resolution,
+            }
+        ),
+        retries=False,
     )
-    resp.raise_for_status()
-    return uuid.UUID(resp.json()["subscription_id"])
+    return uuid.UUID(json.loads(response.data)["subscription_id"])
```
Comment (BYK) [inline src/sentry/incidents/logic.py]: Nit: no more status check?

File: src/sentry/incidents/logic.py [hunk 4]
```diff
@@ -781,5 +789,4 @@ def delete_snuba_subscription(subscription_id):
     :param subscription_id: The uuid of the subscription to delete
     :return:
     """
-    resp = safe_urlopen(settings.SENTRY_SNUBA + "/subscriptions/%s" % subscription_id, "DELETE")
-    resp.raise_for_status()
+    _snuba_pool.urlopen("DELETE", "/subscriptions/%s" % subscription_id, retries=False)
```
Comment (BYK) [inline src/sentry/incidents/logic.py]: Nit: `retries=False` here is redundant as retries only apply to `POST` and `GET`: https://github.com/getsentry/sentry/blob/95767d455b8004ec4b4c5026d84b64b6348e6d37/src/sentry/utils/snuba.py#L241

Comment (mattrobenolt) [reply src/sentry/incidents/logic.py]: We can/should add DELETE to that list too. I only added GET/POST becuase we didn't do any other operations.

**Pull request merged**